themes/default/content/blog/launch-redshift-into-aws-vpc-with-pulumi/index.md

```diff
@@ -0,0 +1,303 @@
+
+To configure the subnet, we provide the Classless Inter-Domain Routing block, which represents the IP range. Then, we associate the subnet with the availability zone (`us-east-1a`) and the subnet with the VPC. Finally, we create the Name tag with the value pulumi-subnet.
+
+To create the subnet group, add the following statements:
+
+```typescript
+// Subnet group
+const subnetGroup = new aws.redshift.SubnetGroup("pulumi-subnet-group", {
+    subnetIds: [
+        subnet.id,
+    ],
+    tags: {
+        Name: "pulumi-subnet-group",
+    },
+});
+```
+
+Under the subnet group, you must provide the collection of subnet identifiers that will join the group. You pass this collection using the `subnetIds` attribute.
+
+Now, update our resource stack by typing:
+
+```bash
+$ pulumi up
+```
+
+The output should look as follows:
+
+![Second update showing the subnet and subnet group being created](./update-2-subnet.png)
+
+You can preview the subnet on the AWS console:
+
+![Screenshot of the AWS console showing the newly provisioned subnet](./aws-console-3-subnet.png)
+
+## Amazon Redshift Cluster
+
+After provisioning the necessary resources, we can move forward to create the Amazon Redshift cluster. To do so, let’s supplement `index.ts` with the following statements:
+
+```typescript
+// Redshift cluster
+const redshiftCluster = new aws.redshift.Cluster("pulumi-redshift-cluster", {
+    clusterIdentifier: "pulumi-redshift-cluster",
+    clusterType: "single-node",
+    databaseName: "tempdb",
+    masterPassword: "Passw0rD",
+    masterUsername: "admin",
+    nodeType: "dc2.large",
+    clusterSubnetGroupName: subnetGroup.name,
+    skipFinalSnapshot: true
+});
+```
+
+We used the hardcoded credentials in the above declaration. In general, the best practice is to [use the Pulumi secret](/docs/intro/concepts/secrets/#secrets).
+
+The above declaration spins up the single-node Redshift cluster using the `dc2.large` node type (2 vCPUs and 15 GB RAM). We also configure the database by setting its name to `tempdb` and the master credentials (username and password) to `admin` and `Passw0rD`, respectively. To pass the name of the subnet group, we use the `clusterSubnetGroupName` attribute. The last parameter, `skipFinalSnapshot` is set to `true`. By doing this, we can delete the cluster without taking its snapshot. The snapshot is useful for recreating the cluster using the previous state. By default, `skipFinalSnapshot` is false, so we have to provide the `finalSnapshotIdentifier`. Otherwise, we can’t delete the cluster.
+
+Now provision the resource:
+
+```bash
+$ pulumi up
+```
+
+The output looks as follows. Note that cluster creation might take up to a few minutes, depending on the AWS region:
+
+![Third update showing the newly provisioned Redshift cluster](./update-2-redshift-cluster.png)
```

Comment:
There does not appear to be an `update-2-redshift-cluster.png` file, and the Hugo preview shows a missing/incorrect image.

Reply:
Ah yep, that should be `3`.